File: atcoder/ABC/095/C-Half-and-Half.py
A, B, C, X, Y = map(int, input().split())
minxy = min(X, Y)
minab = min(A + B, 2 * C)
mina = min(A, 2 * C)
minb = min(B, 2 * C)
X = X - minxy
Y = Y - minxy

ans = minxy * minab + X * mina + Y * minb
print(ans)

File: atcoder/ABC/095/C-Half-and-Half.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int A, B, C, X, Y;
    cin >> A >> B >> C >> X >> Y;
    int minv = min(X, Y);
    int minxy = min(A + B, 2 * C);
    int minx = min(A, 2 * C);
    int miny = min(B, 2 * C);
    X -= minv;
    Y -= minv;

    int ans = minv * minxy + X * minx + Y * miny;
    cout << ans;

    return 0;
}
